### PR TITLE
[9.12.r1] drivers: staging: Port lowmemorykiller to 4.19

### PR DESCRIPTION
**Pull request by sledges**
This is useful for environments where a userspace low memory killer daemon is not available.

Also added Sony's TNG patches (#2106), as well as zram consideration (#2498).

### CODE REVIEW

**Comment by Thaodan** [comment]
Please remove the submodule changes.
Except that it looks quite good so far.

**Review by mkosola — APPROVED** [review]

**Comment by sledges** [comment]
> Please remove the submodule changes.

Removed.

**Review by Thaodan — APPROVED** [review]

File: drivers/staging/android/lowmemorykiller.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*
```
Comment (YumeMichi) [inline drivers/staging/android/lowmemorykiller.h:1]: This file does not belong to this commit.

Comment (sledges) [reply drivers/staging/android/lowmemorykiller.h:1]: Thank you, moved to the next commit.
